### ISSUE TITLE
It should be possible to use Go to Definition from within SDK files

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#151]
I `F12`'d in to the Future class and a request was sent that resulted in this error.

``` text
[18:58:02]: ==> {"id":"14","method":"analysis.getNavigation","params":{"file":"m:\\Apps\\Dart\\lib\\async\\future.dart","offset":525,"length":0}}
[18:58:02]: <== {"error":{"code":"GET_NAVIGATION_INVALID_FILE","message":"Error during `analysis.getNavigation`: invalid file."},"id":"14"}
```

**Comment by DanTup** [comment]
To repro; `F12` onto the `Future` class, put cursor at end of line 95 (the end of the multiline comment ending just before the class def) and when you press `Shift + Ctrl` (which I was doing to start highlighting the comment) the error occurs.

**Comment by DanTup** [comment]
This seems to be because we'er not analyzing the SDK files, so there's no data for this.

Related discussion:

https://groups.google.com/a/dartlang.org/forum/#!topic/analyzer-discuss/_vwPIG4OYco

For now, this doesn't cause any real issues (we only see the error when running in the debug host). Recategorising this as a feature (to support it).

**Comment by DanTup** [comment]
Although this is as-spec'd, it sounds like this is not intended. As a workaround we can possible use the subscription rather than the get request.

**Comment by DanTup** [comment]
After implementing a workaround, raise a case to fix properly in future.

**Comment by DanTup** [comment]
Related SDK issue: https://github.com/dart-lang/sdk/issues/27214

**Comment by DanTup** [comment]
Does not appear fixed in 1.20.1.

**Comment by DanTup** [comment]
Works in 1.23 dev.
